src/mcts/node.cc

```diff
@@ -367,14 +368,23 @@ V4TrainingData Node::GetV4TrainingData(GameResult game_result,
   const auto& position = history.Last();
   const auto& castlings = position.GetBoard().castlings();
   // Populate castlings.
-  result.castling_us_ooo = castlings.we_can_000() ? 1 : 0;
-  result.castling_us_oo = castlings.we_can_00() ? 1 : 0;
-  result.castling_them_ooo = castlings.they_can_000() ? 1 : 0;
-  result.castling_them_oo = castlings.they_can_00() ? 1 : 0;
+  // For non-frc trained nets, just send 1 like we used to.
+  uint8_t queen_side = 1;
+  uint8_t king_side = 1;
+  // If frc trained, send the bit mask representing rook position.
+  if (input_format == pblczero::NetworkFormat::INPUT_112_WITH_CASTLING_PLANE) {
+    queen_side <<= castlings.queenside_rook();
+    king_side <<= castlings.kingside_rook();
+  }
+
+  result.castling_us_ooo = castlings.we_can_000() ? queen_side : 0;
```

Comment:
As we don't store `input_format`, it's not clear from the training data whether FRC planes are in old or new format.

Maybe keep those values as they were and add additional ones? (either 4 values encoded like this or just `castlings.queenside_rook()` and `castlings.kingside_rook()`, or bitmask `(1 << castlings.kingside_rook()) | (1 << castlings.queenside_rook())` although that probably isn't worth that 1 byte).

Reply:
we do store input_format in training data - its new in v5 format.